tasklist/client/src/modules/utils/getQueryVariables.tsx

```diff
@@ -72,95 +49,93 @@ const getQueryVariables = (
     case 'assigned-to-me': {
       return {
         ...BASE_QUERY_VARIABLES,
-        assigned: true,
-        assignee: assignee!,
-        state: 'CREATED',
-        ...parsedFilters,
+        filter: {
+          assignee: currentUserId!,
+          state: 'CREATED',
+          ...parsedFilters,
+        },
       };
     }
     case 'unassigned': {
       return {
         ...BASE_QUERY_VARIABLES,
-        assigned: false,
-        state: 'CREATED',
-        ...parsedFilters,
+        filter: {
+          state: 'CREATED',
+          ...parsedFilters,
+        },
       };
     }
     case 'completed': {
       return {
         ...BASE_QUERY_VARIABLES,
-        state: 'COMPLETED',
-        ...parsedFilters,
+        filter: {
+          state: 'COMPLETED',
+          ...parsedFilters,
+        },
       };
     }
     case 'all-open': {
       return {
         ...BASE_QUERY_VARIABLES,
-        state: 'CREATED',
-        ...parsedFilters,
+        filter: {
+          state: 'CREATED',
+          ...parsedFilters,
+        },
       };
     }
     case 'custom':
     default: {
       return {
         ...BASE_QUERY_VARIABLES,
-        ...parsedFilters,
-        taskVariables,
+        filter:
+          variables.length === 0
+            ? parsedFilters
+            : {
+                ...parsedFilters,
+                variables,
+              },
       };
     }
   }
 };
 
 function convertFiltersToQueryVariables(
   filters: Omit<TaskFilters, 'sortBy' | 'sortOrder'>,
-): TasksSearchBody {
+): NonNullable<QueryUserTasksRequestBody['filter']> {
   const {
     filter,
-    dueDateFrom,
-    dueDateTo,
-    followUpDateFrom,
-    followUpDateTo,
+    candidateGroup,
+    candidateUser,
+    processInstanceKey,
+    processDefinitionKey,
+    userTaskKey,
     ...restFilters
   } = filters;
-  const updatedFilters: TasksSearchBody = restFilters;
+  const numberFilters =
+    numberFiltersSchema.safeParse({
+      processInstanceKey,
+      processDefinitionKey,
+      userTaskKey,
+    }).data ?? {};
+  const updatedFilters: QueryUserTasksRequestBody['filter'] = {
```

Comment:
This will be removed when we migrate to using strings instead of numbers for the ids